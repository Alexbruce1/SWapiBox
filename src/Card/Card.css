
.card {
  /* background-color: #928288; */
  background-color: #f2f2f2;
  border: 5px solid #5468B1;
  border-radius: 12px;
  height: 20rem;
  width: 15rem;
  margin: 5rem;
}

.card h1 {
  color: #5468B1;
}

.card p {
  color: #01236D;
  font-weight: 700;
}