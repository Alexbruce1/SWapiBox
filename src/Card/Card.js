import React, { Component } from 'react';

import './Card.css';
import PropTypes from 'prop-types';

class Card extends Component {
  constructor(props) {
    super(props);

    this.state = {
      favorite: false,
    }
  }

  changeFavorite = () => {
    this.setState({
      favorite: !this.state.favorite
    })
  }

  componentDidUpdate = () => {
    const { Name, population, Species, Homeworld } = this.props.person;
    }

    
    render() {
    const { peoplename, population, species, homeworld } = this.props.people;
    const { planetname, terrain, planetpopulation, climate, residents} = this.props.planets;
    const { vehiclename , model, vehicle_class, passengers } = this.props.vehicles;
    return(
      <div className="person-card" >
        <div className="card-title">
          <h2 className="card-name">
            { peoplename }
          </h2>
        </div>
        <p className="card-species">
          { species }
        </p>
        <p className="card-home">
          <strong>
            home World: 
          </strong>
          { homeworld }
        </p>
        <p className="card-population">
          <strong>
            Population: 
          </strong>
          { population }
        </p>
        <button className="favorite-button" onClick={this.changeFavorite} >Add To Favorites</button>
      </div>
    )
  }
}

Card.propTypes = {
  person: PropTypes.object.isRequired,
}

export default Card;