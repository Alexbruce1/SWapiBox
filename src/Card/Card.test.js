import React from 'react'
import { shallow } from 'enzyme'

import Card from './Card'

describe( 'Card', () => {

  it('Should match the snapshot', () => {
    const mockProp = [{
      homeworld: 'stewjon',
      name: 'Luke',
      species: 'Human',
      population: '2'
      }]

    const wrapper = shallow(<Card {...mockProp} />)
    expect(wrapper).toMatchSnapshot()
  })

  
})