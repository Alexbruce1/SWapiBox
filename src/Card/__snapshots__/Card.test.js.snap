// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    person={
      Object {
        "Homeworld": "stewjon",
        "Name": "Luke",
        "Species": "Human",
        "population": "2",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card-title"
        >
          <h2
            className="card-name"
          >
            Luke
          </h2>
        </div>,
        <p
          className="card-species"
        >
          Human
        </p>,
        <p
          className="card-home"
        >
          <strong>
            Home World:
          </strong>
          stewjon
        </p>,
        <p
          className="card-population"
        >
          <strong>
            Population:
          </strong>
          2
        </p>,
        <button
          className="favorite-button"
        >
          Add To Favorites
        </button>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2
            className="card-name"
          >
            Luke
          </h2>,
          "className": "card-title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Luke",
            "className": "card-name",
          },
          "ref": null,
          "rendered": "Luke",
          "type": "h2",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Human",
          "className": "card-species",
        },
        "ref": null,
        "rendered": "Human",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <strong>
              Home World:
            </strong>,
            "stewjon",
          ],
          "className": "card-home",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Home World:",
            },
            "ref": null,
            "rendered": "Home World:",
            "type": "strong",
          },
          "stewjon",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <strong>
              Population:
            </strong>,
            "2",
          ],
          "className": "card-population",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Population:",
            },
            "ref": null,
            "rendered": "Population:",
            "type": "strong",
          },
          "2",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add To Favorites",
          "className": "favorite-button",
        },
        "ref": null,
        "rendered": "Add To Favorites",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-title"
          >
            <h2
              className="card-name"
            >
              Luke
            </h2>
          </div>,
          <p
            className="card-species"
          >
            Human
          </p>,
          <p
            className="card-home"
          >
            <strong>
              Home World:
            </strong>
            stewjon
          </p>,
          <p
            className="card-population"
          >
            <strong>
              Population:
            </strong>
            2
          </p>,
          <button
            className="favorite-button"
          >
            Add To Favorites
          </button>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2
              className="card-name"
            >
              Luke
            </h2>,
            "className": "card-title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Luke",
              "className": "card-name",
            },
            "ref": null,
            "rendered": "Luke",
            "type": "h2",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Human",
            "className": "card-species",
          },
          "ref": null,
          "rendered": "Human",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                Home World:
              </strong>,
              "stewjon",
            ],
            "className": "card-home",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Home World:",
              },
              "ref": null,
              "rendered": "Home World:",
              "type": "strong",
            },
            "stewjon",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                Population:
              </strong>,
              "2",
            ],
            "className": "card-population",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Population:",
              },
              "ref": null,
              "rendered": "Population:",
              "type": "strong",
            },
            "2",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add To Favorites",
            "className": "favorite-button",
          },
          "ref": null,
          "rendered": "Add To Favorites",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
